woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2117,6 +2118,62 @@ def test_setitem_invalid_input(sample_df):
         df.ww['signup_date'] = df.signup_date
 
 
+def test_setitem_indexed_column_on_unindexed_dataframe(sample_df):
+    sample_df.ww.init()
+
+    col = sample_df.ww.pop('id')
+    col.ww.init(semantic_tags='index')
+
+    warning = "Cannot allow adding column with index tags. Tag removed"
+
+    with pytest.warns(IndexTagRemovedWarning, match=warning):
+        sample_df.ww['id'] = col
+
+    assert sample_df.ww.index is None
+    assert ww.is_schema_valid(sample_df, sample_df.ww.schema)
+    assert sample_df.ww['id'].ww.semantic_tags == {'numeric'}
+
+
+def test_setitem_indexed_column_on_indexed_dataframe(sample_df):
+    sample_df.ww.init()
+    sample_df.ww.set_index('id')
+
+    col = sample_df.ww.pop('id')
+
+    warning = "Cannot allow adding column with index tags. Tag removed"
+
+    with pytest.warns(IndexTagRemovedWarning, match=warning):
+        sample_df.ww['id'] = col
+
+    assert sample_df.ww.index is None
+    assert ww.is_schema_valid(sample_df, sample_df.ww.schema)
+    assert sample_df.ww['id'].ww.semantic_tags == {'numeric'}
+
+    sample_df.ww.init(logical_types={'email': 'Categorical'})
```

Comment:
Defintiely a nitpick, but lets add some whitespace to break up the flow from here to line 2158. the dataframe setup should be one block, then removing the column and setting it up, then when we add it back to the dataframe, and then the assertions